src/inspectorscripts.ts

```diff
@@ -62,29 +87,98 @@ def _jupyterlab_variableinspector_getcontentof(x):
+    if tf and isinstance(x, tf.Variable):
+        # This is to allow tf.Variable.__repr__ to appear
+        return x.__repr__().replace("<", "").replace(">", "").strip()
     return str(x)[:200]
 
 
+def _jupyterlab_variableinspector_is_matrix(x):
+    # True if type(x).__name__ in ["DataFrame", "ndarray", "Series"] else False
+    if pd and isinstance(x, pd.DataFrame):
+        return True
+    if pd and isinstance(x, pd.Series):
+        return True
+    if np and isinstance(x, np.ndarray):
+        return True
+    if pyspark and isinstance(x, pyspark.sql.DataFrame):
+        return True
+    if tf and isinstance(x, tf.Variable):
+        return True
+    return False
+
+
 def _jupyterlab_variableinspector_dict_list():
+    def keep_cond(v):
+        if isinstance(eval(v), str):
+            return True
+        if tf and isinstance(eval(v), tf.Variable):
+            return True
+        if str(eval(v))[0] == "<":
+            return False
+        if  v in ['np', 'pd', 'pyspark', 'tf']:
+            return eval(v) is not None
+        if str(eval(v)).startswith("_Feature"):
+            # removes tf/keras objects
+            return False
+        return True
     values = _jupyterlab_variableinspector_nms.who_ls()
-    vardic = [{'varName': _v, 'varType': type(eval(_v)).__name__, 
+    vardic = [{'varName': _v, 
+    'varType': type(eval(_v)).__name__, 
     'varSize': str(_jupyterlab_variableinspector_getsizeof(eval(_v))), 
     'varShape': str(_jupyterlab_variableinspector_getshapeof(eval(_v))) if _jupyterlab_variableinspector_getshapeof(eval(_v)) else '', 
     'varContent': str(_jupyterlab_variableinspector_getcontentof(eval(_v))), 
-    'isMatrix': True if type(eval(_v)).__name__ in ["DataFrame", "ndarray", "Series"] else False}
-            for _v in values if ((str(eval(_v))[0] != "<") or (isinstance(eval(_v), str)))]
+    'isMatrix': _jupyterlab_variableinspector_is_matrix(eval(_v))}
+            for _v in values if keep_cond(_v)]
     return json.dumps(vardic)
 
+
 def _jupyterlab_variableinspector_getmatrixcontent(x):
-    if np and pd and type(x).__name__ in ["Series", "DataFrame"]:
+    def get_np_threshold():
+        if np and np.get_printoptions()['threshold'] != 1000: # this is the default
+            return np.get_printoptions()['threshold']
+        return _jupyterlab_variableinspector_maxrows
+    
+    def get_pd_threshold():
+        if pd and pd.get_option('max_rows') != 60:
+            return pd.get_option('max_rows')
+        return _jupyterlab_variableinspector_maxrows
+    
+    def get_threshold(np_thres, pd_thres):
+        if np_thres is None:
+            return pd_thres
+        elif pd_thres is None:
+            return np_thres
+        elif np_thres < 0 or pd_thres < 0:
+            return None
+        else:
+            return max(np_thres, pd_thres)
+        
+    np_threshold = get_np_threshold()
+    pd_threshold = get_pd_threshold()
+    threshold = get_threshold(np_threshold, pd_threshold)
```

Comment:
What do you think about if we just use our `_jupyterlab_variableinspector_maxrows` here instead of checking the np and pd thresholds. I think the settings on the kernel should not affect the behaviour of the extension if possible.

Reply:
I'm okay with this.